pkg/scheduler/framework/plugins/podtopologyspread/filtering.go

```diff
@@ -309,8 +319,21 @@ func (pl *PodTopologySpread) Filter(ctx context.Context, cycleState *framework.C
 			continue
 		}
 		// judging criteria:
-		// 'existing matching num' + 'if self-match (1 or 0)' - 'global min matching num' <= 'maxSkew'
+		// 'existing matching num' + 'if self-match (1 or 0)' - 'global minimum' <= 'maxSkew'
 		minMatchNum := paths[0].MatchNum
+
+		domainsNum, ok := s.TpKeyToDomainsNum[c.TopologyKey]
+		if !ok {
+			// error which should not happen.
+			// If this error occurs, MinDomains will be ignored in judge.
+			klog.ErrorS(nil, "Internal error occurred while retrieving the number of eligible domains from topology key. MinDomains will be ignored.", "topologyKey", tpKey, "domainsNums", s.TpKeyToDomainsNum)
```

Comment:
`domainsNum` here will always be 0, the same as `minMatchNum`. Change to something like:
```golang
		if domainsNum, ok := s.TpKeyToDomainsNum[c.TopologyKey]; ok {
			if domainsNum < c.MinDomains {
				minMatchNum = 0
			}
		} else {
			// When the number of eligible domains with matching topology keys is less than `minDomains`,
			// it treats "global minimum" as 0.
			klog.ErrorS(nil, "Internal error occurred while retrieving the number of eligible domains from topology key. MinDomains will be ignored.", "topologyKey", tpKey, "domainsNums", s.TpKeyToDomainsNum)
		}
```

Reply:
Thanks for the nice catch 🙏